src/requests/consumption.js

```diff
@@ -0,0 +1,377 @@
+				{ _id: templateId },
+				{
+					$set: {
+						tasks: taskCreationResponse.taskIds,
+						taskSequence: taskCreationResponse.externalIds,
+					},
+				}
+			)
+
+			//update the published id in resource table
+			await resourceService.publishCallback(templateData.id, templateId.toString())
+
+			//return result
+			result.success = true
+			result.templateId = templateId
+			return resolve(result)
+		} catch (error) {
+			result.error = `Error: ${error.message}`
+			return reject(error)
+		}
+	})
+}
+
+/**
+ * Format Project Template
+ * @name formatTemplate
+ * @param {Object} templateData - Project template data
+ * @returns {Object} - Response contains formatted template
+ */
+const formatTemplate = (templateData) => {
+	try {
+		let template = {
+			title: templateData.title,
+			description: templateData.objective || '',
+			keywords: templateData.keywords ? templateData.keywords.split(',').map((k) => k.trim()) : [],
+			isDeleted: false,
+			recommendedFor: templateData.recommended_for?.length
+				? templateData.recommended_for.map((item) => item.label)
+				: [],
+			createdBy: templateData.user_id,
+			updatedBy: templateData.user_id,
+			learningResources: templateData.learning_resources ? convertResources(templateData.learning_resources) : [],
+			isReusable: true,
+			taskSequence: [], // Initially empty
+			deleted: false,
+			status: common.PUBLISHED_STATUS,
+			externalId: generateExternalId(templateData.title),
+			entityType: '',
+			metaInformation: {
+				goal: '',
+				rationale: '',
+				primaryAudience: '',
+				duration: `${templateData.recommended_duration.number} ${templateData.recommended_duration.duration}`,
+				successIndicators: '',
+				risks: '',
+				approaches: '',
+			},
+			tasks: [], // Initially empty
+		}
+
+		return { success: true, template }
+	} catch (error) {
+		console.error('Error in formatTemplate:', error.message)
+		return { success: false, error: error.message }
+	}
+}
+
+/**
+ * Create and Find Categories
+ * @name processCategories
+ * @param {Object} categories - Categories Data
+ * @returns {Object} - Response contains categories data
+ */
+async function processCategories(categories) {
+	try {
+		const categoriesCollection = mongoDb.collection(COLLECTIONS.CATEGORIES)
+
+		// Format categories
+		const formattedCategories = categories.map((category) => {
+			if (!category.label || !category.value) {
+				throw new Error('Each category must have a label and a value.')
+			}
+			return {
+				label: category.label,
+				value: category.value,
+				formattedName: formatCategoriesName(category.value),
+				externalId: category.value.replace(/_/g, '').toLowerCase(),
+			}
+		})
+
+		// Fetch existing categories by externalId
+		const existingCategories = await categoriesCollection
+			.find({ externalId: { $in: formattedCategories.map((cat) => cat.externalId) } })
+			.toArray()
+		const existingExternalIds = new Set(existingCategories.map((cat) => cat.externalId))
```

Comment:
Can we convert it to array? 
`[ ...new Set(existingCategories.map((cat) => cat.externalId))]`
or are we intending to use it as Set only? 
@priyanka-TL